modules/puzzle/src/main/PuzzleTheme.scala

```diff
@@ -7,77 +7,80 @@ case class PuzzleTheme(key: PuzzleTheme.Key, name: I18nKey, description: I18nKey
+  val mateIn5        = PuzzleTheme(i.mateIn5, i.mateIn5Description)
+  val smotheredMate  = PuzzleTheme(i.smotheredMate, i.smotheredMateDescription)
+  val middlegame     = PuzzleTheme(i.middlegame, i.middlegameDescription)
+  val oneMove        = PuzzleTheme(i.oneMove, i.oneMoveDescription)
+  val opening        = PuzzleTheme(i.opening, i.openingDescription)
+  val pawnEndgame    = PuzzleTheme(i.pawnEndgame, i.pawnEndgameDescription)
+  val pin            = PuzzleTheme(i.pin, i.pinDescription)
+  val promotion      = PuzzleTheme(i.promotion, i.promotionDescription)
+  val queenEndgame   = PuzzleTheme(i.queenEndgame, i.queenEndgameDescription)
   val queenRookEndgame =
-    PuzzleTheme(Key("queenRookEndgame"), i.queenRookEndgame, i.queenRookEndgameDescription)
-  val queensideAttack = PuzzleTheme(Key("queensideAttack"), i.queensideAttack, i.queensideAttackDescription)
-  val quietMove       = PuzzleTheme(Key("quietMove"), i.quietMove, i.quietMoveDescription)
-  val rookEndgame     = PuzzleTheme(Key("rookEndgame"), i.rookEndgame, i.rookEndgameDescription)
-  val sacrifice       = PuzzleTheme(Key("sacrifice"), i.sacrifice, i.sacrificeDescription)
-  val short           = PuzzleTheme(Key("short"), i.short, i.shortDescription)
-  val skewer          = PuzzleTheme(Key("skewer"), i.skewer, i.skewerDescription)
-  val superGM         = PuzzleTheme(Key("superGM"), i.superGM, i.superGMDescription)
-  val trappedPiece    = PuzzleTheme(Key("trappedPiece"), i.trappedPiece, i.trappedPieceDescription)
-  val underPromotion  = PuzzleTheme(Key("underPromotion"), i.underPromotion, i.underPromotionDescription)
-  val veryLong        = PuzzleTheme(Key("veryLong"), i.veryLong, i.veryLongDescription)
-  val xRayAttack      = PuzzleTheme(Key("xRayAttack"), i.xRayAttack, i.xRayAttackDescription)
-  val zugzwang        = PuzzleTheme(Key("zugzwang"), i.zugzwang, i.zugzwangDescription)
+    PuzzleTheme(i.queenRookEndgame, i.queenRookEndgameDescription)
+  val queensideAttack = PuzzleTheme(i.queensideAttack, i.queensideAttackDescription)
+  val quietMove       = PuzzleTheme(i.quietMove, i.quietMoveDescription)
+  val rookEndgame     = PuzzleTheme(i.rookEndgame, i.rookEndgameDescription)
+  val sacrifice       = PuzzleTheme(i.sacrifice, i.sacrificeDescription)
+  val short           = PuzzleTheme(i.short, i.shortDescription)
+  val skewer          = PuzzleTheme(i.skewer, i.skewerDescription)
+  val superGM         = PuzzleTheme(i.superGM, i.superGMDescription)
+  val trappedPiece    = PuzzleTheme(i.trappedPiece, i.trappedPieceDescription)
+  val underPromotion  = PuzzleTheme(i.underPromotion, i.underPromotionDescription)
+  val veryLong        = PuzzleTheme(i.veryLong, i.veryLongDescription)
+  val xRayAttack      = PuzzleTheme(i.xRayAttack, i.xRayAttackDescription)
+  val zugzwang        = PuzzleTheme(i.zugzwang, i.zugzwangDescription)
   val checkFirst      = PuzzleTheme(Key("checkFirst"), I18nKey("Check first"), I18nKey("Check first"))
```

Comment:
should we go on and create the checkFirst key so that PuzzleTheme construction interface is uniform?

Reply:
Because it's never displayed, I would say no. I prefer having a puzzle key that is not a translation, rather than having a translation that is not meant to ever be displayed.